src/core/thread/mesh_forwarder.cpp

```diff
@@ -571,6 +595,9 @@ Message *MeshForwarder::PrepareNextDirectTransmission(void)
         switch (error)
         {
         case kErrorNone:
+#if OPENTHREAD_CONFIG_TX_QUEUE_STATISTICS_ENABLE
+            mTxQueueStats.UpdateFor(*curMessage);
+#endif
```

Comment:
This doesn't capture the time spent actually trying to deliver the message. Should we be recording the time in something like `UpdateSendMessage()`?

Reply:
Yes. But that's intentional. The idea is to use collected time-in-queue histogram info to validate/tune parameters for "delay-aware queue management" (which uses this the time message is in queue till it is prepared to be sent).